django/utils/http.py

```diff
@@ -366,3 +366,35 @@ def escape_leading_slashes(url):
     if url.startswith("//"):
         url = "/%2F{}".format(url[2:])
     return url
+
+
+def _parseparam(s):
+    while s[:1] == ";":
+        s = s[1:]
+        end = s.find(";")
+        while end > 0 and (s.count('"', 0, end) - s.count('\\"', 0, end)) % 2:
+            end = s.find(";", end + 1)
+        if end < 0:
+            end = len(s)
+        f = s[:end]
+        yield f.strip()
+        s = s[end:]
+
+
+def parse_header_parameters(line):
+    """Parse a Content-type like header.
+    Return the main content-type and a dictionary of options.
+    """
+    parts = _parseparam(";" + line)
+    key = parts.__next__()
+    pdict = {}
+    for p in parts:
+        i = p.find("=")
+        if i >= 0:
+            name = p[:i].strip().lower()
+            value = p[i + 1 :].strip()
+            if len(value) >= 2 and value[0] == value[-1] == '"':
+                value = value[1:-1]
+                value = value.replace("\\\\", "\\").replace('\\"', '"')
```

Comment:
This branch in untested :thinking:

Reply:
Added test cases from https://github.com/python/cpython/blob/d8cf3514dd4682419a66f6e834bb384ee34afc95/Lib/test/test_cgi.py#L526-L553